systems.py

```diff
@@ -5,31 +5,34 @@
 
 class Systems():
   
+  # Creates and initializes all of the controllers
   def __init__(self):
     self.controllers = [];
-    controllers[0] = MotorController
-    controllers[1] = BatteryController
-    controllers[2] = ClimateController
-    controllers[3] = SensorController
-    controllers[4] = BackupController
+    controllers[0] = MotorController()
+    controllers[1] = BatteryController()
+    controllers[2] = ClimateController()
+    controllers[3] = SensorController()
+    controllers[4] = BackupController()
 
+  # Finds and returns the value of the specified variable in a certain controller
   def get(name: str) -> any:
     for i in controllers:
-      if name in controllers[i].variables:
-        return controllers[i].variables[name].value
-    return null
+      value = controllers[i].getVariable(controllers[i], name)
+      return value
```

Comment:
Line 21: By my eye, if one iteration of the outer for loop is completed and the desired variable is not found, value will be undeclared and likely raise an exception when returned on line 21. Your initial code with return null makes sense to me but in Python the equivalent is None. Furthermore, just double check your indentation to ensure None is returned at the right time.